File: sample/src/main/java/me/panpf/sketch/sample/net/GiphyService.java
package me.panpf.sketch.sample.net;

import me.panpf.sketch.sample.bean.GiphySearchResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GiphyService {

    @GET("/v1/gifs/search?type=gifs&sort=&api_key=Gc7131jiJuvI7IdN0HZ1D7nh0ow5BU6g&pingback_id=17c5f87f46b18d99")
    Call<GiphySearchResponse> search(
            @Query("q") String queryWord,
            @Query("offset") int pageStart,
            @Query("limit") int pageSize);
}

File: sample/src/main/java/com/github/panpf/sketch/sample/net/GiphyService.kt
package com.github.panpf.sketch.sample.net

import retrofit2.http.GET
import com.github.panpf.sketch.sample.bean.GiphySearchResponse
import retrofit2.Response
import retrofit2.http.Query

interface GiphyService {
    @GET("/v1/gifs/search?type=gifs&sort=&api_key=Gc7131jiJuvI7IdN0HZ1D7nh0ow5BU6g&pingback_id=17c5f87f46b18d99")
    suspend fun search(
        @Query("q") queryWord: String?,
        @Query("offset") pageStart: Int,
        @Query("limit") pageSize: Int
    ): Response<GiphySearchResponse>
}